File: app/src/androidTest/java/com/example/bernatvarela/espressotestapp/presentation/tests/BaseTest.java
package com.example.bernatvarela.espressotestapp.presentation.tests;

import android.support.test.rule.ActivityTestRule;

import com.example.bernatvarela.espressotestapp.MainActivity;
import com.example.bernatvarela.espressotestapp.presentation.espresso.EspressoBase;

import org.junit.Rule;

public class BaseTest {
    protected EspressoBase espresso;

    public BaseTest() {
        espresso = EspressoBase.getInstance();
    }

    @Rule
    public ActivityTestRule<MainActivity> mActivityRule = new ActivityTestRule(MainActivity.class);
}

File: app/src/androidTest/java/com/example/bernatvarela/espressotestapp/presentation/tests/BaseTest.kt
package com.example.bernatvarela.espressotestapp.presentation.tests

import android.support.test.rule.ActivityTestRule

import com.example.bernatvarela.espressotestapp.MainActivity
import com.example.bernatvarela.espressotestapp.presentation.espresso.EspressoBase

import org.junit.Rule

open class BaseTest {
    protected var espresso: EspressoBase

    init {
        espresso = EspressoBase()
    }

    @JvmField
    @Rule
    var mActivityRule: ActivityTestRule<MainActivity> = ActivityTestRule(MainActivity::class.java)
}
